agent/util-scripts/pbench-tool-meister-start

```diff
@@ -0,0 +1,542 @@
+    """
+    try:
+        with pid_file.open("r") as fp:
+            raw_pid = fp.read()
+    except Exception:
+        # No "pid" to kill
+        return 2
+    else:
+        try:
+            pid = int(raw_pid)
+        except Exception:
+            # Bad pid value
+            return 3
+        try:
+            os.kill(pid, signal.SIGKILL)
+        except OSError as exc:
+            if exc.errno == errno.ESRCH:
+                # PID not found, ignore
+                return 4
+            else:
+                # Some error encountered trying to KILL the process.
+                return 5
+        except Exception:
+            # Some other error encountered trying to KILL the process.
+            return 6
+        else:
+            # "successfully" KILL'd the give process.
+            return 1
+
+
+def main(argv):
+    """Main program for the tool meister start.
+    """
+    _prog = Path(argv[0])
+    _prog_dir = _prog.parent
+    PROG = _prog.name
+    logger = logging.getLogger(PROG)
+    if os.environ.get("_PBENCH_TOOL_MEISTER_START_LOG_LEVEL") == "debug":
+        log_level = logging.DEBUG
+    else:
+        log_level = logging.INFO
+    logger.setLevel(log_level)
+    sh = logging.StreamHandler()
+    sh.setLevel(log_level)
+    shf = logging.Formatter("%(message)s")
+    sh.setFormatter(shf)
+    logger.addHandler(sh)
+
+    try:
+        group = argv[1]
+    except IndexError:
+        group = "default"
+
+    # 1. Load the tool group data given the tool group argument
+    try:
+        tool_group = ToolGroup(group)
+    except Exception:
+        logger.exception("failed to load tool group data")
+        return 1
+
+    try:
+        benchmark_run_dir = os.environ["benchmark_run_dir"]
+        hostname = os.environ["hostname"]
+        full_hostname = os.environ["full_hostname"]
+    except Exception:
+        logger.exception("failed to fetch parameters from the environment")
+        return 1
+    else:
+        tm_dir = Path(benchmark_run_dir, "tm")
+        try:
+            os.mkdir(tm_dir)
+            os.chdir(tm_dir)
+        except Exception:
+            logger.exception("failed to create the local tool meister directory")
+            return 1
+    if not full_hostname or not hostname:
+        logger.error(
+            "ERROR - hostname ('%s') and full_hostname ('%s') environment"
+            " variables are required",
+            hostname,
+            full_hostname,
+        )
+        return 1
+    if os.environ.get("_PBENCH_UNIT_TESTS"):
+        # FIXME: this is an artifact of the unit test environment.
+        hostnames = "localhost"
+    else:
+        hostnames = f"localhost {full_hostname}"
+    params = {"hostnames": hostnames, "tm_dir": tm_dir, "redis_port": redis_port}
+
+    # 2. Start the Redis Server (config of port from agent config)
+    #   - the Redis Server is requested to create the PID file
+
+    # Create the Redis Server pbench-specific configuration file
```

Comment:
fixme: jinja2 template and let the users supply their own if needed.

Reply:
Not convinced we want to do this first round.  Let's leave this as an option somebody can implement later.